replicator/replicator.go

```diff
@@ -311,13 +319,22 @@ func config_replica(m_ctx context.Context, m_cli immuclient.ImmuClient, r_ctx co
 	}
 	settings := settings_resp.Settings
 	replication_settings := schema.ReplicationNullableSettings{
-		Replica:          &schema.NullableBool{Value: true},
-		MasterDatabase:   &schema.NullableString{Value: src_db},
-		MasterAddress:    &schema.NullableString{Value: config.MasterAddr},
-		MasterPort:       &schema.NullableUint32{Value: uint32(config.MasterPort)},
-		FollowerUsername: &schema.NullableString{Value: config.FollowerUsername},
-		FollowerPassword: &schema.NullableString{Value: config.FollowerPassword},
+		Replica:                      &schema.NullableBool{Value: true},
+		MasterDatabase:               &schema.NullableString{Value: src_db},
+		MasterAddress:                &schema.NullableString{Value: config.MasterAddr},
+		MasterPort:                   &schema.NullableUint32{Value: uint32(config.MasterPort)},
+		FollowerUsername:             &schema.NullableString{Value: config.FollowerUsername},
+		FollowerPassword:             &schema.NullableString{Value: config.FollowerPassword},
+		PrefetchTxBufferSize:         &schema.NullableUint32{Value: uint32(config.PrefetchTxBufferSize)},
+		ReplicationCommitConcurrency: &schema.NullableUint32{Value: uint32(config.ReplicationCommitConcurrency)},
 	}
+
+	// SyncAcks > 0 implies this is a database with synchronous replication enabled
```

Comment:
Question: if I understand it correctly, activating a replica of a database which *has sync replication enabled* will always create a synchronous replica. 

If it is true, then we need a flag to disable this. Even better, I'd like to have an explicit flag for telling the replicator if the replica is going to create need to be synchronous or not.

That is because we use an *asynchronous* replica for creating backups. Since the backup operation can take a lot of time, if we configure it a synchronous, it would block the master during backup and it will slow it down during regular operations

Reply:
@SimoneLazzaris Thank you for your comment, are you suggesting I add an explicit flag to enable syncing of databases which have synchronous replication enabled?

Reply:
Since these are per database settings, I thought if a database on master has synchronous replication enabled, then automatically the database should be created on the replica machine. Am I missing something in my understanding here?